NuRadioReco/modules/channelCWNotchFilter.py

```diff
@@ -63,8 +63,40 @@ def find_frequency_peaks(freq: np.ndarray, spectrum : np.ndarray, threshold : fl
 
     return freq[peak_idxs]
 
+def get_filter(freq : int, quality_factor=1e3, fs=3.2e9 * units.Hz, cache=None):
+    """
+    Function to get single notch filter for a given frequency
+    Parameters
+    ----------
+    freq : np.ndarray
+        Frequency
+    quality_factor : int, default = 1000
+        quality factor of the notch filter, defined as the ratio f0/bw, where f0 is the centre frequency
+        and bw the bandwidth of the filter at (f0,-3 dB)
+    fs : float, default = 3.2e9 Hz
+        sampling frequency
+    cache : dict, default = None,
+        Optional caching dictionary. The function will check whether the frequency to be filtered
+        is in the dictionary values and will otherwise add it
+        !!! Note this does not cache the quality factor information
+    Returns
+    -------
+        filter : list, shape (6)
+            second order IIR notch filter at frequency freq
+    """
+    if cache is not None:
+        if freq in cache.keys():
+            return cache[freq]
+    filter = signal.iirnotch(freq, quality_factor, fs=fs)
+    if cache is not None:
+        # Check to avoid cache dictionary overflowing the memory,
+        # set to roughly stay below 6 MB (every filter is 6 floats + freq = 7 floats ~ 56B)
+        if len(cache.keys() < 1e5):
```

Comment:
does the filter depend on the number of samples of the frequency spectrum?

Reply:
I guess that is handled by the sampling rate which reflects that. I would opt again of removing the default value. Because you might what to upsample the trace before for more accuracy but forget to adjust the sampling rate.